railties/lib/rails/commands/server/server_command.rb

```diff
@@ -232,6 +241,13 @@ def pid
         def self.banner(*)
           "rails server [puma, thin etc] [options]"
         end
+
+        def prepare_restart
+          return unless File.exist?("tmp/run_restart.txt")
+
+          FileUtils.rm_f(server_options[:pid])
+          FileUtils.rm_f("tmp/run_restart.txt")
```

Comment:
Never mind. We pass a `restart_cmd` to Puma, so the `rails server` *is* executed on restart.